Common/src/Pollster/AgentHandler.cs

```diff
@@ -43,31 +43,47 @@ namespace ArmoniK.Core.Common.Pollster;
 /// </summary>
 public class AgentHandler : IAgentHandler, IAsyncDisposable
 {
-  private readonly WebApplication        app_;
-  private readonly ComputePlane          computePlaneOptions_;
-  private readonly ILogger<AgentHandler> logger_;
-  private readonly IObjectStorageFactory objectStorageFactory_;
-  private readonly GrpcAgentService      service_;
-  private readonly ISubmitter            submitter_;
+  private readonly WebApplication                       app_;
+  private readonly ComputePlane                         computePlaneOptions_;
+  private readonly Injection.Options.DependencyResolver dependencyResolverOptions_;
+  private readonly ILogger<AgentHandler>                logger_;
+  private readonly IObjectStorageFactory                objectStorageFactory_;
+  private readonly IPushQueueStorage                    pushQueueStorage_;
+  private readonly IResultTable                         resultTable_;
+  private readonly GrpcAgentService                     service_;
+  private readonly ISubmitter                           submitter_;
+  private readonly ITaskTable                           taskTable_;
 
   /// <summary>
   ///   Initializes a new instance
   /// </summary>
   /// <param name="loggerInit">Logger initializer used to configure the loggers needed by the worker</param>
   /// <param name="computePlaneOptions">Options needed for the creation of the servers</param>
+  /// <param name="dependencyResolverOptions">Configuration for DependencyResolver</param>
   /// <param name="submitter">Interface to manage tasks</param>
   /// <param name="objectStorageFactory">Interface class to create object storage</param>
+  /// <param name="pushQueueStorage">Interface to put tasks in the queue</param>
+  /// <param name="taskTable">Interface to manage task states</param>
+  /// <param name="resultTable">Interface to manage result states</param>
```

Comment:
That begins to be too many params, just for the sake of being passed to the Agent. Couldn't we rely on the dependency injection to build the Agent?

(Could be done in another PR)

Reply:
With the way it is created, I don't know if we can do it easily with DI. Moreover, we need a new `Agent` for every task. The `AgentHandler` is already created with DI.